include/CoreGraphics/CGContext.h

```diff
@@ -261,4 +261,22 @@ COREGRAPHICS_EXPORT CGPoint CGContextConvertPointToUserSpace(CGContextRef c, CGP
 COREGRAPHICS_EXPORT CGSize CGContextConvertSizeToDeviceSpace(CGContextRef c, CGSize size);
 COREGRAPHICS_EXPORT CGSize CGContextConvertSizeToUserSpace(CGContextRef c, CGSize size);
 COREGRAPHICS_EXPORT CGRect CGContextConvertRectToDeviceSpace(CGContextRef c, CGRect rect);
-COREGRAPHICS_EXPORT CGRect CGContextConvertRectToUserSpace(CGContextRef c, CGRect rect);
\ No newline at end of file
+COREGRAPHICS_EXPORT CGRect CGContextConvertRectToUserSpace(CGContextRef c, CGRect rect);
+
+// [WinObjC Extension]
+// CGContextIwEnableEnhancedErrorHandling allows a developer to turn on advanced context error
+// handling. A CGContext that encounters errors when rendering will, by default, assert and abort.
+// Enhanced handling both silences asserts and enables an interested consumer to determine whether a
+// context has encountered a fatal drawing error.
+// A developer using enhanced error handling MUST check the status of a completed set of drawing
+// operations using CGContextIwGetError() and determine whether to redraw the frame.
+COREGRAPHICS_EXPORT void CGContextIwEnableEnhancedErrorHandling(CGContextRef context);
+
+COREGRAPHICS_EXPORT const CFStringRef kCGErrorDomainIslandwood;
+
+typedef CF_ENUM(CFIndex, CGContextIwErrorCode) {
+    kCGContextErrorDeviceReset,
+    kCGContextErrorInvalidParameter,
+};
+
+COREGRAPHICS_EXPORT bool CGContextIwGetError(CGContextRef context, CFErrorRef* /* returns-retained */ error);
```

Comment:
>/* returns-retained */ [](start = 79, length = 22)

since this is a public api and we are adding it, we should add c/c++ style document to it, to explicitly say what it returns and what's going on.